packages/api/src/controllers/clip.ts

```diff
@@ -160,4 +165,100 @@ app.post("/", validatePost("clip-payload"), async (req, res) => {
   });
 });
 
+const fieldsMap = {
+  id: `asset.ID`,
+  createdAt: { val: `asset.data->'createdAt'`, type: "int" },
+  updatedAt: { val: `asset.data->'status'->'updatedAt'`, type: "int" },
+  userId: `asset.data->>'userId'`,
+  playbackId: `asset.data->>'playbackId'`,
+  sourceType: `asset.data->'source'->>'type'`,
+  sourceSessionId: `asset.data->'source'->>'sessionId'`,
+  sourceAssetId: `asset.data->'source'->>'assetId'`,
+} as const;
+
+app.get("/:id", authorizer({}), async (req, res) => {
+  const { id } = req.params;
+
+  let { limit, cursor, all, allUsers, order, filters, count, cid, ...otherQs } =
+    toStringValues(req.query);
+
+  if (!order) {
+    order = "updatedAt-true,createdAt-true";
+  }
+
+  const query = [...parseFilters(fieldsMap, filters)];
+
+  if (!req.user.admin || !all || all === "false") {
+    query.push(sql`asset.data->>'deleted' IS NULL`);
+  }
+
+  let output: WithID<Asset>[];
+  let newCursor: string;
+
+  query.push(sql`asset.data->>'userId' = ${req.user.id}`);
+
+  const content =
+    (await db.asset.getByPlaybackId(id)) ||
+    (await db.session.get(id)) ||
+    (await db.asset.get(id)) ||
+    (await db.stream.getByIdOrPlaybackId(id));
+
+  if (!content) {
+    throw new NotFoundError("Content not found");
+  }
+
+  let contentType: string;
+
+  if ("objectStoreId" in content) {
+    contentType = "asset";
+  } else if ("streamKey" in content) {
+    contentType = "stream";
+  } else {
+    contentType = "session";
+  }
+
+  query.push(sql`asset.data->'source'->>'type' = 'clip'`);
+  switch (contentType) {
+    case "asset":
+      query.push(sql`asset.data->'source'->>'assetId' = ${content.id}`);
+      break;
+    case "session":
+      query.push(sql`asset.data->'source'->>'sessionId' = ${content.id}`);
+      break;
+    case "stream":
+      const sessionQuery = [];
+      sessionQuery.push(sql`data->>'parentId' = ${content.id}`);
+      const [sessions] = await db.session.find(sessionQuery, {
+        order: `data->>'lastSeen' DESC NULLS LAST`,
+        cursor,
+      });
```

Comment:
This should be a temp hack, there might be a _ton_ (thousands) of streams here for any long living stream (say, lot radio 🛸 ). proper fix should involve a join (on the regular asset list query there's a join with the `user` table, check it out)